Using Procfile.dev tries to parse as a Proxfile

Hi there,

Just tried to start up prox using a file called Procfile.dev and I get the following output:

```
$ prox --procfile Procfile.dev
[ERROR] Failed to parse Procfile: failed to decode Proxfile as YAML: yaml: unmarshal errors:
  line 1: field web not found in type prox.Proxfile
  line 2: field worker not found in type prox.Proxfile
```

--- a/cmd/prox/_testdata/Procfile
+++ b/cmd/prox/_testdata/Procfile
@@ -1,0 +1,3 @@
+redis: bin/redis-server conf/redis.conf
+server: php -S localhost:8080 app/web/index.php
+selenium: java -jar /usr/local/bin/selenium-server-standalone.jar

--- a/cmd/prox/_testdata/Procfile.dev
+++ b/cmd/prox/_testdata/Procfile.dev
@@ -1,0 +1,1 @@
+Procfile

--- a/cmd/prox/_testdata/Proxfile
+++ b/cmd/prox/_testdata/Proxfile
@@ -1,0 +1,4 @@
+processes:
+  redis: bin/redis-server conf/redis.conf
+  server:   "php -S localhost:8080 app/web/index.php"
+  selenium: "java -jar /usr/local/bin/selenium-server-standalone.jar"

--- a/cmd/prox/_testdata/default-to-Procfile/Procfile
+++ b/cmd/prox/_testdata/default-to-Procfile/Procfile
@@ -1,0 +1,1 @@
+../Procfile

--- a/cmd/prox/_testdata/default-to-Proxfile/Proxfile
+++ b/cmd/prox/_testdata/default-to-Proxfile/Proxfile
@@ -1,0 +1,1 @@
+../Proxfile

--- a/cmd/prox/main.go
+++ b/cmd/prox/main.go
@@ -4,9 +4,11 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"io"
 	"os"
 	"os/signal"
 	"path/filepath"
+	"strings"
 	"syscall"
 
 	"github.com/fgrosse/prox"
@@ -92,37 +94,39 @@ func environment(path string) (prox.Environment, error) {
 	return env, err
 }
 
-func processes(env prox.Environment, procFileFlag string) ([]prox.Process, error) {
-	var (
-		f     *os.File
-		err   error
-		parse = prox.ParseProxFile
-	)
+func processes(env prox.Environment, path string) ([]prox.Process, error) {
+	var parser func(io.Reader, prox.Environment) ([]prox.Process, error)
 
-	if procFileFlag != "" {
-		// user has specified a path
-		logger.Debug("Reading processes from file specified via --procfile", zap.String("path", procFileFlag))
-		f, err = os.Open(procFileFlag)
-		if filepath.Base(procFileFlag) == "Procfile" {
-			parse = prox.ParseProcFile
-		}
-	} else {
-		// default to "Proxfile"
-		f, err = os.Open("Proxfile")
-		if os.IsNotExist(err) {
-			// no "Proxfile" but maybe we can open a "Procfile"
-			logger.Debug("Reading processes from Procfile")
-			parse = prox.ParseProcFile
-			f, err = os.Open("Procfile")
+	switch {
+	case path == "" && fileExists("Proxfile"):
+		logger.Debug("Reading processes from Proxfile")
+		path = "Proxfile"
+		parser = prox.ParseProxFile
+	case path == "" && fileExists("Procfile"):
+		logger.Debug("Reading processes from Procfile")
+		path = "Procfile"
+		parser = prox.ParseProcFile
+	case path != "":
+		logger.Debug("Reading processes from file specified via --procfile", zap.String("path", path))
+		if strings.HasPrefix(filepath.Base(path), "Procfile") {
+			parser = prox.ParseProcFile
 		} else {
-			logger.Debug("Reading processes from Proxfile")
+			parser = prox.ParseProxFile
 		}
+	default:
+		return nil, errors.New("no Proxfile or Procfile file found. Please specify a path with --procfile")
 	}
 
+	f, err := os.Open(path)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to open process file: %w", err)
 	}
 
 	defer f.Close()
-	return parse(f, env)
+	return parser(f, env)
+}
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
 }

--- a/cmd/prox/main_test.go
+++ b/cmd/prox/main_test.go
@@ -1,12 +1,13 @@
 package main
 
 import (
+	"os"
+	"testing"
+
 	"github.com/fgrosse/prox"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap/zaptest"
-	"os"
-	"testing"
 )
 
 func TestParseProcessesFile(t *testing.T) {

--- a/cmd/prox/main_test.go
+++ b/cmd/prox/main_test.go
@@ -1,0 +1,55 @@
+package main
+
+import (
+	"github.com/fgrosse/prox"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+	"os"
+	"testing"
+)
+
+func TestParseProcessesFile(t *testing.T) {
+	cases := map[string]struct {
+		dir  string
+		path string
+	}{
+		"default Procfile": {dir: "_testdata/default-to-Procfile"},
+		"default Proxfile": {dir: "_testdata/default-to-Proxfile"},
+		"Proxfile":         {dir: "_testdata", path: "Proxfile"},
+		"Procfile":         {path: "_testdata/Procfile"},
+		"Procfile.dev":     {path: "_testdata/Procfile.dev"},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			logger = zaptest.NewLogger(t)
+			env := prox.Environment{}
+
+			if c.dir != "" {
+				changeDirectory(t, c.dir)
+			}
+
+			pp, err := processes(env, c.path)
+			require.NoError(t, err)
+			assert.Len(t, pp, 3)
+		})
+	}
+}
+
+func changeDirectory(t *testing.T, dir string) {
+	t.Helper()
+
+	originalDir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Failed to get current directory: %v", err)
+	}
+
+	err = os.Chdir(dir)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		err := os.Chdir(originalDir)
+		require.NoError(t, err)
+	})
+}

